server/datastore/mysql/labels_test.go

```diff
@@ -788,6 +790,10 @@ func setupLabelSpecsTest(t *testing.T, ds fleet.Datastore) []*fleet.LabelSpec {
 	err := ds.ApplyLabelSpecs(context.Background(), expectedSpecs)
 	require.Nil(t, err)
 
+	// Because `Hosts` for manual labels matches both host name AND host ID,
+	// specifying "1" will match both host with ID 1 (whose name is "0")
+	// and host with name "1".
+	expectedSpecs[4].Hosts = []string{"0", "1", "2", "3", "4"}
 	return expectedSpecs
```

Comment:
I could have updated the test so that the host names have letters in them, but decided to keep them as-is to reinforce that we can now have collisions between ID and host name, serial or uuid.